modules/stanford_events_importer/src/Plugin/Field/FieldWidget/StanfordEventsImporterAPIURLFieldWidget.php

```diff
@@ -0,0 +1,238 @@
+ *   }
+ * )
+ */
+class StanfordEventsImporterAPIURLFieldWidget extends LinkWidget {
+
+  /**
+   * Path to feed xml.
+   */
+  const XML_FEED = "https://events.stanford.edu/xml/drupal/v2.php";
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+
+    // De-falt.
+    $element = parent::formElement($items, $delta, $element, $form, $form_state);
+    $element['#attached']['library'][] = 'core/drupal.states';
+
+    /** @var \Drupal\link\LinkItemInterface $item */
+    $item = $items[$delta];
+    $defaults = $this->parseURLForDefaults($item);
+
+    // Add a type of feed.
+    $element['_other']['type'] = [
+      '#type' => 'select',
+      '#title' => $this->t("Event Group Option"),
+      '#options' => [
+        '' => $this->t("- Select Option -"),
+        'organization' => $this->t("Organization"),
+        'category' => $this->t("Category"),
+        'featured' => $this->t('Featured'),
+        'today' => $this->t('Today'),
+      ],
+      '#default_value' => $defaults['type'] ?? '',
+    ];
+
+    // The organization id (integer) as provided in the xml feed.
+    $element['_other']['organization'] = [
+      '#type' => 'select',
+      '#title' => $this->t("Organization"),
+      '#field_parents' => ['other'],
+      '#options' => $this->getOrgOptions(),
+      '#states' => [
+        'visible' => [
+          '#edit-su-event-xml-url-' . $delta . '-other-type' => ['value' => 'organization'],
+        ],
+      ],
+      '#default_value' => $defaults['organization'] ?? '',
+    ];
+
+    $element['_other']['category'] = [
+      '#type' => 'select',
+      '#title' => $this->t("Category"),
+      '#field_parents' => ['other'],
+      '#options' => $this->getCatOptions(),
+      '#states' => [
+        'visible' => [
+          '#edit-su-event-xml-url-' . $delta . '-other-type' => ['value' => 'category'],
+        ],
+      ],
+      '#default_value' => $defaults['category'] ?? '',
+    ];
+
+    $element['_other']['org_status'] = [
+      '#type' => 'select',
+      '#title' => $this->t("Event Status"),
+      '#field_parents' => ['other'],
+      '#options' => [
+        'published' => $this->t("Published"),
+        'unlisted' => $this->t("Unlisted"),
+        'bookmarked' => $this->t("Bookmarked"),
+      ],
+      '#states' => [
+        'visible' => [
+          '#edit-su-event-xml-url-' . $delta . '-other-type' => ['value' => 'organization'],
+        ],
+      ],
+      '#default_value' => $defaults['org_status'] ?? '',
+    ];
+
+    // Hide the uri.
+    $element['uri']['#type'] = "hidden";
+    $element['uri']['#required'] = FALSE;
+    return $element;
+  }
+
+  /**
+   * Get a key/value list of organizations from the API.
+   *
+   * @return array
+   *   An array of select form options.
+   */
+  protected function getOrgOptions() {
+    $opts = \Drupal::state()->get("stanford_events_importer_orgs");
```

Comment:
no `\Drupal` in classes. Also i think this would be best as cache. then it'll clean up automatically at some time.